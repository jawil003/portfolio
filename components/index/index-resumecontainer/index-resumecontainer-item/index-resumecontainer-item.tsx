import React from "react";
import designSystem from "../../../../styles/designSystem";
import { Spacer } from "../../../shared/spacer/spacer";
import { Typography } from "../../../shared/typography";

export interface Props {
  title: string;
  description: string;
  latest?: boolean;
  start_year: number | null;
  end_year: number | null;
}

/**
 * An ResumeItem React Component.
 * @author Jannik Will
 * @version 0.1
 */
export const IndexResumeContainerItem: React.FC<Props> = ({
  description,
  title,
  start_year,
  end_year,
  latest,
}) => {
  return (
    <div
      style={{
        position: "relative",
        display: "flex",
        minHeight: "200px",
      }}
    >
      <div
        style={{
          display: "flex",
          flex: 1,
          justifyContent: "center",
        }}
      >
        <div
          style={{
            backgroundColor:
              designSystem.colors
                .palette.white.dark,
            width: "3.5px",
            height: latest
              ? "calc(100% + 70px)"
              : "100%",
          }}
        />
      </div>
      <div
        style={{
          position: "absolute",
          top: 0,
          left: 0,
          width: "100%",
          height: "100%",
          display: "flex",
        }}
      >
        <div
          style={{
            flex: 1,
            display: "grid",
            justifyContent: "flex-end",
            alignItems: "flex-start",
            alignContent: "flex-start",
            color:
              designSystem.colors.brand
                .primary,
          }}
        >
          {start_year ? (
            [
              <Typography
                key="start_year"
                align="center"
                variant="h6"
                color={
                  designSystem.colors
                    .brand.primary
                }
              >
                {start_year}
              </Typography>,
              <Typography
                key="end_year"
                align="center"
                variant="h6"
                color={
                  designSystem.colors
                    .brand.primary
                }
              >
                {end_year || "heute"}
              </Typography>,
            ]
          ) : (
            <Typography
              key="end_year"
              align="center"
              variant="h6"
              color={
                designSystem.colors
                  .brand.primary
              }
            >
              {end_year}
            </Typography>
          )}
        </div>
        <div
          style={{
            width: "25px",
            display: "flex",
            flexDirection: "column",
            justifyContent:
              "flex-start",
            margin: "0px 20px",
          }}
        >
          <div
            style={{
              height: "25px",
              border: `5px solid ${designSystem.colors.brand.primary}`,
              borderRadius: "100%",
              backgroundColor:
                designSystem.colors
                  .brand.secondary,
            }}
          />
        </div>
        <div
          style={{
            flex: 1,
            display: "flex",
            flexDirection: "column",
            justifyContent:
              "flex-start",
            alignItems: "flex-start",
            color:
              designSystem.colors.brand
                .primary,
          }}
        >
          <Typography
            maxLength="100px"
            bold="semi-bold"
            color={
              designSystem.colors.brand
                .primary
            }
            variant="h6"
          >
            {title}
          </Typography>
          <Spacer height="0.4em" />
          <Typography
            maxLength="150px"
            variant="b2"
            color={
              designSystem.colors.brand
                .primary
            }
            italic
          >
            {description}
          </Typography>
        </div>
      </div>
    </div>
  );
};

IndexResumeContainerItem.defaultProps = {
  latest: false,
};
