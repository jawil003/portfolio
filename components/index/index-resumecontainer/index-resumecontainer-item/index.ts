export * from "./index-resumecontainer-item";