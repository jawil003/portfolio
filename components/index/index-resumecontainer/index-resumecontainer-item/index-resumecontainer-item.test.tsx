import { render } from "@testing-library/react";
import {IndexResumeContainerItem, Props} from "./index-resumecontainer-item";

const variants: Props[] = [
  {
    description: "Test",
    end_year: 2020,
    title: "Test",
    start_year: 2019,
  },
  {
    description: "Test",
    end_year: null,
    title: "Test",
    start_year: 2019,
  },
  {
    description: "Test",
    end_year: 2020,
    title: "Test",
    start_year: null,
  },
];

describe("IndexResumeContainerItem", ()=>{
    test.each(variants)(
      "Create IndexResumeContainerItem with %p", (variant)=>{
          const testItem = render(
            <IndexResumeContainerItem {...variant}/>,
          );

          expect(testItem.baseElement).toMatchSnapshot();
      }
    );
})