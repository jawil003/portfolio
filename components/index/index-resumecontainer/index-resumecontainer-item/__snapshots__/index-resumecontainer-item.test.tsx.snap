// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexResumeContainerItem Create IndexResumeContainerItem with {"description": "Test", "end_year": 2020, "start_year": 2019, "title": "Test"} 1`] = `
.emotion-0 {
  text-align: center;
  font-style: normal;
  margin: unset;
  -webkit-text-decoration: none;
  text-decoration: none;
  letter-spacing: normal;
  word-spacing: normal;
  color: #f3f3f3;
  word-wrap: unset;
  padding: unset;
  word-break: normal;
}

.emotion-0 {
  font-weight: 400;
  font-size: 1.25em;
}

.emotion-4 {
  text-align: start;
  font-style: normal;
  margin: unset;
  -webkit-text-decoration: none;
  text-decoration: none;
  max-width: 100px;
  letter-spacing: normal;
  word-spacing: normal;
  color: #f3f3f3;
  word-wrap: unset;
  padding: unset;
  word-break: normal;
}

.emotion-4 {
  font-weight: 600;
  font-size: 1.25em;
}

.emotion-6 {
  text-align: start;
  font-style: italic;
  margin: unset;
  -webkit-text-decoration: none;
  text-decoration: none;
  max-width: 150px;
  letter-spacing: normal;
  word-spacing: normal;
  color: #f3f3f3;
  word-wrap: unset;
  padding: unset;
  word-break: normal;
}

.emotion-6 {
  font-weight: 400;
  font-size: 0.875em;
}

<body>
  <div>
    <div
      style="position: relative; display: flex; min-height: 200px;"
    >
      <div
        style="display: flex; flex: 1; justify-content: center;"
      >
        <div
          style="background-color: rgb(240, 240, 240); width: 3.5px; height: 100%;"
        />
      </div>
      <div
        style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; display: flex;"
      >
        <div
          style="flex: 1; display: grid; justify-content: flex-end; align-items: flex-start; align-content: flex-start; color: rgb(243, 243, 243);"
        >
          <h6
            class="emotion-0 emotion-1"
            color="#f3f3f3"
            id="heading6"
          >
            2019
          </h6>
          <h6
            class="emotion-0 emotion-1"
            color="#f3f3f3"
            id="heading6"
          >
            2020
          </h6>
        </div>
        <div
          style="width: 25px; display: flex; flex-direction: column; justify-content: flex-start; margin: 0px 20px;"
        >
          <div
            style="height: 25px; border: 5px solid #f3f3f3; border-radius: 100%; background-color: rgb(0, 115, 157);"
          />
        </div>
        <div
          style="flex: 1; display: flex; flex-direction: column; justify-content: flex-start; align-items: flex-start; color: rgb(243, 243, 243);"
        >
          <h6
            class="emotion-4 emotion-1"
            color="#f3f3f3"
            id="heading6"
            maxlength="100px"
          >
            Test
          </h6>
          <div
            style="height: 0.4em;"
          />
          <p
            class="emotion-6 emotion-7"
            color="#f3f3f3"
            id="body2"
            maxlength="150px"
          >
            Test
          </p>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`IndexResumeContainerItem Create IndexResumeContainerItem with {"description": "Test", "end_year": 2020, "start_year": null, "title": "Test"} 1`] = `
.emotion-0 {
  text-align: center;
  font-style: normal;
  margin: unset;
  -webkit-text-decoration: none;
  text-decoration: none;
  letter-spacing: normal;
  word-spacing: normal;
  color: #f3f3f3;
  word-wrap: unset;
  padding: unset;
  word-break: normal;
}

.emotion-0 {
  font-weight: 400;
  font-size: 1.25em;
}

.emotion-2 {
  text-align: start;
  font-style: normal;
  margin: unset;
  -webkit-text-decoration: none;
  text-decoration: none;
  max-width: 100px;
  letter-spacing: normal;
  word-spacing: normal;
  color: #f3f3f3;
  word-wrap: unset;
  padding: unset;
  word-break: normal;
}

.emotion-2 {
  font-weight: 600;
  font-size: 1.25em;
}

.emotion-4 {
  text-align: start;
  font-style: italic;
  margin: unset;
  -webkit-text-decoration: none;
  text-decoration: none;
  max-width: 150px;
  letter-spacing: normal;
  word-spacing: normal;
  color: #f3f3f3;
  word-wrap: unset;
  padding: unset;
  word-break: normal;
}

.emotion-4 {
  font-weight: 400;
  font-size: 0.875em;
}

<body>
  <div>
    <div
      style="position: relative; display: flex; min-height: 200px;"
    >
      <div
        style="display: flex; flex: 1; justify-content: center;"
      >
        <div
          style="background-color: rgb(240, 240, 240); width: 3.5px; height: 100%;"
        />
      </div>
      <div
        style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; display: flex;"
      >
        <div
          style="flex: 1; display: grid; justify-content: flex-end; align-items: flex-start; align-content: flex-start; color: rgb(243, 243, 243);"
        >
          <h6
            class="emotion-0 emotion-1"
            color="#f3f3f3"
            id="heading6"
          >
            2020
          </h6>
        </div>
        <div
          style="width: 25px; display: flex; flex-direction: column; justify-content: flex-start; margin: 0px 20px;"
        >
          <div
            style="height: 25px; border: 5px solid #f3f3f3; border-radius: 100%; background-color: rgb(0, 115, 157);"
          />
        </div>
        <div
          style="flex: 1; display: flex; flex-direction: column; justify-content: flex-start; align-items: flex-start; color: rgb(243, 243, 243);"
        >
          <h6
            class="emotion-2 emotion-1"
            color="#f3f3f3"
            id="heading6"
            maxlength="100px"
          >
            Test
          </h6>
          <div
            style="height: 0.4em;"
          />
          <p
            class="emotion-4 emotion-5"
            color="#f3f3f3"
            id="body2"
            maxlength="150px"
          >
            Test
          </p>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`IndexResumeContainerItem Create IndexResumeContainerItem with {"description": "Test", "end_year": null, "start_year": 2019, "title": "Test"} 1`] = `
.emotion-0 {
  text-align: center;
  font-style: normal;
  margin: unset;
  -webkit-text-decoration: none;
  text-decoration: none;
  letter-spacing: normal;
  word-spacing: normal;
  color: #f3f3f3;
  word-wrap: unset;
  padding: unset;
  word-break: normal;
}

.emotion-0 {
  font-weight: 400;
  font-size: 1.25em;
}

.emotion-4 {
  text-align: start;
  font-style: normal;
  margin: unset;
  -webkit-text-decoration: none;
  text-decoration: none;
  max-width: 100px;
  letter-spacing: normal;
  word-spacing: normal;
  color: #f3f3f3;
  word-wrap: unset;
  padding: unset;
  word-break: normal;
}

.emotion-4 {
  font-weight: 600;
  font-size: 1.25em;
}

.emotion-6 {
  text-align: start;
  font-style: italic;
  margin: unset;
  -webkit-text-decoration: none;
  text-decoration: none;
  max-width: 150px;
  letter-spacing: normal;
  word-spacing: normal;
  color: #f3f3f3;
  word-wrap: unset;
  padding: unset;
  word-break: normal;
}

.emotion-6 {
  font-weight: 400;
  font-size: 0.875em;
}

<body>
  <div>
    <div
      style="position: relative; display: flex; min-height: 200px;"
    >
      <div
        style="display: flex; flex: 1; justify-content: center;"
      >
        <div
          style="background-color: rgb(240, 240, 240); width: 3.5px; height: 100%;"
        />
      </div>
      <div
        style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; display: flex;"
      >
        <div
          style="flex: 1; display: grid; justify-content: flex-end; align-items: flex-start; align-content: flex-start; color: rgb(243, 243, 243);"
        >
          <h6
            class="emotion-0 emotion-1"
            color="#f3f3f3"
            id="heading6"
          >
            2019
          </h6>
          <h6
            class="emotion-0 emotion-1"
            color="#f3f3f3"
            id="heading6"
          >
            heute
          </h6>
        </div>
        <div
          style="width: 25px; display: flex; flex-direction: column; justify-content: flex-start; margin: 0px 20px;"
        >
          <div
            style="height: 25px; border: 5px solid #f3f3f3; border-radius: 100%; background-color: rgb(0, 115, 157);"
          />
        </div>
        <div
          style="flex: 1; display: flex; flex-direction: column; justify-content: flex-start; align-items: flex-start; color: rgb(243, 243, 243);"
        >
          <h6
            class="emotion-4 emotion-1"
            color="#f3f3f3"
            id="heading6"
            maxlength="100px"
          >
            Test
          </h6>
          <div
            style="height: 0.4em;"
          />
          <p
            class="emotion-6 emotion-7"
            color="#f3f3f3"
            id="body2"
            maxlength="150px"
          >
            Test
          </p>
        </div>
      </div>
    </div>
  </div>
</body>
`;
